### PR TITLE
Improve lsp spec part 2

### PR DESCRIPTION
**Pull request by peterfication**

### CODE REVIEW

**Comment by peterfication** [comment]
@johmsalas I tried your approach with waiting for the LSP outside the `it` statement and it seems to work fine.

I don't really think we need the additional wait in between test cases because we wait before the assert already, right?

File: tests/textcase/lsp/lsp_spec.lua [hunk 1]
```diff
@@ -44,5 +44,18 @@ describe("LSP", function()
       local expected_code = test_helpers.read_file("./tests/textcase/lsp/fixtures/snake-case.ts")
       assert.are.same(table.concat(content, "\n"), expected_code)
     end)
+
+    it("Should be triggered on keybinding for constant case", function()
+      test_helpers.execute_keys("/variable_to_be_tested<CR>gaN")
```
Comment (johmsalas) [inline tests/textcase/lsp/lsp_spec.lua:49]: Should the document be reloaded to keep the tests isolated?

File: tests/textcase/lsp/lsp_spec.lua [hunk 2]
```diff
@@ -44,5 +44,18 @@ describe("LSP", function()
...
+      local content = nil
+      test_helpers.wait_for(5 * 1000, function()
```
Comment (johmsalas) [inline tests/textcase/lsp/lsp_spec.lua:51]: This await check is a good candidate to become a function (in the future). [WET](https://kentcdodds.com/blog/aha-programming) is totally fine

**Review by johmsalas — APPROVED** [review]

File: tests/textcase/lsp/lsp_spec.lua (hunk 1 above)
Comment (peterfication) [reply tests/textcase/lsp/lsp_spec.lua:49]: Very important point! Thanks for pointing it out :)

File: tests/textcase/lsp/lsp_spec.lua (hunk 2 above)
Comment (peterfication) [reply tests/textcase/lsp/lsp_spec.lua:51]: True!

**Comment by peterfication** [comment]
In the end, the LS startup can/should live in the `before_each` method. In the end, it won't be started twice as it keeps running the background (I assume). About the type definition issue you mentioned in the other PR: it seems to work fine. The main issue seems to be the first loading.

However, `act` is not happy with matrix testing. It probably takes too many resources at the same time. If I have just one Neovim version in the ci.yml, it works.

**Review by johmsalas — APPROVED** [review]
LSP tests got enough care for now 🍾 

Let me know if act keeps showing issues. I'm not sure if .env is a thing in Lua but we can figure out a solution. I'll also monitored it

**Pull request merged**